packages/cli/src/lib/certs/flags.ts

```diff
@@ -1,44 +1,55 @@
-import {all} from './endpoints'
+import {SniEndpoint} from '../types/sni_endpoint'
 import {APIClient} from '@heroku-cli/command'
+import {HTTP} from 'http-call'
+import {Domain} from '../types/domain'
 
 export default async function (flags: { endpoint: string | undefined; name: string | undefined; app: string }, heroku: APIClient) {
   if (flags.endpoint && flags.name) {
     throw new Error('Specified both --name and --endpoint, please use just one')
   }
 
-  let endpoints = await all(flags.app, heroku)
+  let {body: sniEndpoints} = await heroku.get<SniEndpoint[]>(`/apps/${flags.app}/sni-endpoints`)
 
-  if (endpoints.length === 0) {
+  if (sniEndpoints.length === 0) {
     throw new Error(`${flags.app} has no SSL certificates`)
   }
 
   if (flags.endpoint) {
-    endpoints = endpoints.filter(function (endpoint) {
-      return endpoint.cname === flags.endpoint
+    const promises: Promise<HTTP<Domain>>[] = []
+    sniEndpoints.forEach(endpoint => {
+      endpoint.domains.forEach(domain => promises.push(heroku.get<Domain>(`/apps/${flags.app}/domains/${domain}`)))
     })
 
-    if (endpoints.length > 1) {
+    const domains = (await Promise.all(promises)).map(({body: domain}) => domain)
+
+    sniEndpoints = sniEndpoints.filter(endpoint => {
+      // This was modified from `endpoint.cname === flags.endpoint` because `cname` doesn't exist anymore in the SniEndpoint serialization.
+      // We're making the assumption that the `--endpoint` flag was being used to search by hostname (internationalized domain name).
+      return domains.some(domain => domain.hostname === flags.endpoint && domain.sni_endpoint?.name === endpoint.name)
+    })
+
+    if (sniEndpoints.length > 1) {
```

Comment:
Passing the flag `--endpoint` used to work matching the provided value against the `cname` property (that now has been removed). The idea of that flag was to match the hostname associated with the SNI Endpoint, but that information is available in the associated `domain` records, so we fetch them and look for a match there.